### PR TITLE
Issue #5450: add javadoc recognition to DesignForExtension

### PR DESCRIPTION
**Pull request by rnveach**
Issue #5450

Regression to come.

### CODE REVIEW

**Comment by rnveach** [comment]
TC did not run in CI.

Regression: http://rveach.no-ip.org/checkstyle/regression/reports/262/

**Comment by romani** [comment]
@pbludov , please help with first review, I have too much assignments now.

**Comment by rnveach** [comment]
@romani asignee approved, it is back to you again.

**Comment by romani** [comment]
@pbludov, please reassign to next reviewer your self , you should have permission

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/design/designforextension/InputDesignForExtensionOverridableMethods.java [hunk 1]
```diff
@@ -147,7 +147,7 @@ public int foo30() {
             return 1;
         }
 
-        /* */
+        /* */ // violation
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/design/designforextension/InputDesignForExtensionOverridableMethods.java]: Please add test where emplty javadoc and non empty block comment a next to each other.
Javadoc above block comment.
Javadoc below block comment.

**Comment by rnveach** [comment]
> http://rveach.no-ip.org/checkstyle/regression/reports/262/Orekit/index.html#A2 is it not a javadoc ? A bit misplaced but is javadoc tool ignore it too ?

I confirmed locally, Javadoc tool will ignore it. This is why we created InvalidJavadocPositionCheck.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/design/designforextension/InputDesignForExtensionOverridableMethods.java (hunk 1 above)
Comment (rnveach) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/design/designforextension/InputDesignForExtensionOverridableMethods.java]: Done.

**Comment by rnveach** [comment]
Note this branch was 4 commits out of date with master.
